specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2021-09-01-preview/appplatform.json

```diff
@@ -3695,7 +3707,62 @@
           },
           "readOnly": true
         }
-      }
+      },
+      "discriminator": "type"
+    },
+    "KeyVaultCertificateProperties": {
+      "description": "Properties of certficate imported from key vault.",
+      "required": [
+        "type",
+        "vaultUri",
+        "keyVaultCertName"
+      ],
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/CertificateProperties"
+        }
+      ],
+      "properties": {
+        "vaultUri": {
+          "description": "The vault uri of user key vault.",
+          "type": "string"
+        },
+        "keyVaultCertName": {
+          "description": "The certificate name of key vault.",
+          "type": "string"
+        },
+        "certVersion": {
+          "description": "The certificate version of key vault.",
+          "type": "string"
+        },
+        "importPrivateKey":{
```

Comment:
Let's change this property to another name and make its default value to be false, since it is a practice to make boolean with default of false. Something like "public key only" = "false", please come up some better name.